webapp/lib/view.dart

```diff
@@ -192,3 +179,135 @@ class AvailablePackagesViewPartial {
+    );
+  }
+
+  DivElement addTagDropDown(List<String> tags) {
+    var addTagModal = new DivElement()
+      ..classes.add('configure-package__response-add-tag-modal');
+    addTagModal.append(
+      HeadingElement.h6()
+      ..classes.add('configure-package__response-add-tag-modal-heading')
+      ..text = 'Select new tag to add');
+    var tagOptions = new SelectElement()
+      ..classes.add('configure-package__response-add-tag-modal-dropdown')
+      ..onChange.listen((event) {
+        var selectedOption = (event.currentTarget as SelectElement).value;
+        controller.command(controller.UIAction.addConfigurationTag, new controller.ConfigurationTagData(tagToAdd: selectedOption));
+        if(tagListElement.children.last is DivElement) tagListElement.children.removeLast();
+      });
+    tagOptions.add(new OptionElement()..text = '--Select Tag--', false);
+    tags.forEach((tag) {
+      var option = new OptionElement()
+        ..text = tag
+        ..value = tag;
+      tagOptions.add(option, false);
+    });
+    addTagModal.append(tagOptions);
+    return addTagModal;
+  }
+}
+
+class ConfigurationViewTagResponsesPartial {
+  DivElement tagResponsesElement;
+  DivElement _tagResponsesHeader;
+  DivElement _tagResponsesBody;
+
+  ConfigurationViewTagResponsesPartial() {
+    tagResponsesElement = new DivElement()
+      ..classes.add('configure-package__tag-responses');
+    _tagResponsesHeader = new DivElement()
+      ..classes.add('configure-package__tag-responses-header');
+    _tagResponsesBody = new DivElement()
+      ..classes.add('configure-package__tag-responses-body');
+  }
+
+  void renderResponses(String tag, Map<String, List<String>> responses) {
+    tagResponsesElement.children.clear();
+    _tagResponsesHeader.children.clear();
+    _tagResponsesBody.children.clear();
+    responses.forEach((language, responseSet) {
+      _tagResponsesHeader.append(new HeadingElement.h5()..text = language);
+      var body = new DivElement()..classes.add('configure-package__tag-responses-body-items');
+      int responseIndex = 0;
+      for (var response in responseSet) {
+        body.append(new ParagraphElement()
+        ..classes.add('configure-package__tag-responses-body-item')
+        ..attributes.addAll({'contenteditable': 'true', 'parent-tag': tag, 'language': '$language' ,'index': '$responseIndex'})
+        ..text = response
+        ..onBlur.listen((event) {
+          var reponseElement = (event.currentTarget as Element);
+          var parentTag = reponseElement.attributes['parent-tag'];
+          var editedResponse = {'index': reponseElement.attributes['index'], 'language': reponseElement.attributes['language'] ,'text': reponseElement.text};
+          controller.command(controller.UIAction.editConfigurationTagResponse, new controller.ConfigurationResponseData(parentTag: parentTag, editedResponse: editedResponse));
```

Comment:
would it make sense to make `index`, `language` and `text` properties of `controller.ConfigurationResponseData`, alongside `parentTag`?

Reply:
yes...I think that's a better way. Changed